### PR TITLE
Unit tests for MQTTPushNotificationProvider

### PR DESCRIPTION
**Pull request by GDLMadushanka**
## Automation tests
 - Unit tests 
   > 80%, 70%

### CODE REVIEW

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt/src/test/resources/conf/carbon.xml [hunk 1]
```diff
@@ -0,0 +1,656 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+
+<!--
+  ~ Copyright 2005-2017 WSO2 Inc. (http://wso2.com)
```
Comment (Megala21) [inline components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt/src/test/resources/conf/carbon.xml]: License header should have only the current year

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt/src/test/resources/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Megala21) [inline components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt/src/test/resources/log4j.properties]: It should be 2017

**Review by Megala21 — CHANGES_REQUESTED** [review]
Since you are adding new test cases for mqtt, shall we add the exec file path to the pom of test-coverage

**Review by Megala21 — APPROVED** [review]

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt/src/test/java/org/wso2/carbon/device/mgt/extensions/push/notification/provider/mqtt/MQTTBasedPushNotificationProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import org.wso2.carbon.event.output.adapter.core.exception.OutputEventAdapterException;
+import org.wso2.carbon.event.output.adapter.core.internal.CarbonOutputEventAdapterService;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/*
+    Unit tests for MQTTBasedPushNotificationProvider class
```
Comment (pamod) [inline components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt/src/test/java/org/wso2/carbon/device/mgt/extensions/push/notification/provider/mqtt/MQTTBasedPushNotificationProviderTest.java:39]: Extra space ?

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt/src/test/java/org/wso2/carbon/device/mgt/extensions/push/notification/provider/mqtt/MQTTNotificationStrategyTest.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        mqttNotificationStrategy = Mockito.mock(MQTTNotificationStrategy.class, Mockito.CALLS_REAL_METHODS);
+        carbonOutputEventAdapterService = Mockito.mock(CarbonOutputEventAdapterService.class,
+                Mockito.CALLS_REAL_METHODS);
+        Mockito.doReturn(true).when(carbonOutputEventAdapterService).isPolled(Mockito.any());
+        Mockito.doNothing().when(carbonOutputEventAdapterService).publish(Mockito.any(), Mockito.any(), Mockito.any());
+        Mockito.doNothing().when(carbonOutputEventAdapterService).destroy(ADAPTER_NAME);
+    }
+
+    private void initializeCarbonContext() throws IOException, RegistryException {
+        if (System.getProperty("carbon.home") == null) {
```
Comment (pamod) [inline components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt/src/test/java/org/wso2/carbon/device/mgt/extensions/push/notification/provider/mqtt/MQTTNotificationStrategyTest.java:76]: This method is duplicated  with MQTTBasedPushNotificationProviderTest#initializeCarbonContext() can't we introduce a util for this ?

File: components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt/src/test/java/org/wso2/carbon/device/mgt/extensions/push/notification/provider/mqtt/MQTTNotificationStrategyTest.java [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+            File file = new File("src/test/resources");
+            if (file.exists()) {
+                System.setProperty("carbon.home", file.getAbsolutePath());
```
Comment (pamod) [inline components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.push.notification.provider.mqtt/src/test/java/org/wso2/carbon/device/mgt/extensions/push/notification/provider/mqtt/MQTTNotificationStrategyTest.java:79]: Can we define a constant for "carbon.home" ?

**Pull request merged**